src/main/java/com/socialmedia/service/PostService.java

```diff
@@ -72,4 +73,28 @@ public Post delete(Long id) {
         .sorted((p1, p2) -> (int) (p1.getDate() - p2.getDate()))
         .collect(Collectors.toList());
   }
+
+  public void updateLikes(Long postId, ApplicationUser author) {
+    AtomicBoolean isPresent = new AtomicBoolean(false);
+
+    Post post = jpaRepository.getOne(postId);
+    List<ApplicationUser> likes = post.getLikes();
+
+    likes.forEach(like -> {
+      if (like.getUsername().equals(author.getUsername())) {
```

Comment:
same efficiency problem, you are iterating whole list though you just need the first match. I suggest to use likes.stream().
filter(like->like.getUsername().equals(author.getUsername())).
findFirst()

You'll get an optional as a result, so you can use it further if (result.isPresent()) ... and you won't need atomic boolean at all